wrappers/c/zxing.cpp

```diff
@@ -0,0 +1,161 @@
+{
+	return iv->height();
+}
+
+int zxing_ImageView_pixStride (zxing_ImageView * iv)
+{
+	return iv->pixStride();
+}
+
+int zxing_ImageView_rowStride (zxing_ImageView * iv)
+{
+	return iv->rowStride();
+}
+
+zxing_ImageFormat zxing_ImageView_format (zxing_ImageView * iv)
+{
+	return (zxing_ImageFormat) iv->format();
+}
+
+/*
+ * ...
+ */
+
+/*
+ * ZXing/BarcodeFormat.h
+ */
+
+zxing_BarcodeFormat zxing_BarcodeFormatFromString (const char * format)
+{
+	return (zxing_BarcodeFormat) ZXing::BarcodeFormatFromString(std::string(format));
+}
+
+// TODO:
+//zxing_BarcodeFormat zxing_BarcodeFormatsFromString (const char * formats)
+//{
+//    return (zxing_BarcodeFormat) ZXing::BarcodeFormatsFromString(std::string(formats));
+//}
+
+/*
+ * ZXing/DecodeHints.h
+ */
+
+zxing_DecodeHints * zxing_DecodeHints_new ()
+{
+	return new ZXing::DecodeHints();
+}
+
+void zxing_DecodeHints_delete (zxing_DecodeHints * hints)
+{
+	delete hints;
+}
+
+void zxing_DecodeHints_setTryHarder (zxing_DecodeHints * hints, bool tryHarder)
+{
+	hints->setTryHarder(tryHarder);
+}
+
+void zxing_DecodeHints_setTryDownscale (zxing_DecodeHints * hints, bool tryDownscale)
+{
+	hints->setTryDownscale(tryDownscale);
+}
+
+void zxing_DecodeHints_setFormats (zxing_DecodeHints * hints, zxing_BarcodeFormat formats)
+{
+	hints->setFormats((ZXing::BarcodeFormat) formats);
+}
+
+void zxing_DecodeHints_setBinarizer (zxing_DecodeHints * hints, zxing_Binarizer binarizer)
+{
+	hints->setBinarizer((ZXing::Binarizer) binarizer);
+}
+
+void zxing_DecodeHints_setEanAddOnSymbol (zxing_DecodeHints * hints, zxing_EanAddOnSymbol eanAddOnSymbol)
+{
+	hints->setEanAddOnSymbol((ZXing::EanAddOnSymbol) eanAddOnSymbol);
+}
+
+void zxing_DecodeHints_setTextMode (zxing_DecodeHints * hints, zxing_TextMode textMode)
+{
+	hints->setTextMode((ZXing::TextMode) textMode);
+}
+
+/*
+ * ...
+ */
+
+/*
+ * ZXing/Result.h
+ */
+
+void zxing_Result_delete (zxing_Result * result)
+{
+	delete result;
+}
+
+bool zxing_Result_isValid (zxing_Result * result)
+{
+	return result->isValid();
+}
+
+zxing_BarcodeFormat zxing_Result_format (zxing_Result * result)
+{
+	return (zxing_BarcodeFormat) result->format();
+}
+
+/*
+ * ...
+ */
+
+/*
+ * ZXing/ReadBarcode.h
+ */
+
+zxing_Result * zxing_ReadBarcode (const zxing_ImageView * iv, const zxing_DecodeHints * hints)
+{
+	const ZXing::ImageView * cppiv = reinterpret_cast<const ZXing::ImageView *>(iv);
+	const ZXing::DecodeHints * cpphints = reinterpret_cast<const ZXing::DecodeHints *>(hints);
+	ZXing::Result result = ZXing::ReadBarcode(*cppiv, *cpphints);
+	return new ZXing::Result(result);
+}
+
+zxing_Results * zxing_ReadBarcodes (const zxing_ImageView * iv, const zxing_DecodeHints * hints)
+{
+	const ZXing::ImageView * cppiv = reinterpret_cast<const ZXing::ImageView *>(iv);
+	const ZXing::DecodeHints * cpphints = reinterpret_cast<const ZXing::DecodeHints *>(hints);
+	ZXing::Results results = ZXing::ReadBarcodes(*cppiv, *cpphints);
+	return new ZXing::Results(results);
```

Comment:
either use std::move or a pass a temporary (which is implicitly an r-value reference). I suggest:
```
return new ZXing::Results(ZXing::ReadBarcodes(*cppiv, *cpphints));
```

Reply:
Oh I misunderstood your other comment. Fixed it now, thanks.